lib/Doctrine/ODM/MongoDB/Mapping/PropertyAccessors/PropertyAccessorFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Mapping\PropertyAccessors;
+
+use ReflectionProperty;
+
+use const PHP_VERSION_ID;
+
+class PropertyAccessorFactory
```

Comment:
This is not annotated as an internal class.

```suggestion
/** @internal */
class PropertyAccessorFactory
```

Reply:
It might be necessary for external package: https://github.com/doctrine-extensions/DoctrineExtensions/blob/eb53dfcb2b592327b76ac5226fbb003d32aea37e/src/Tree/Strategy/ORM/Closure.php#L138-L142

Reply:
I'd mark it as internal so we can then remove it before the next major version. I'd still like to see this logic either in doctrine/persistence, doctrine/reflection, or a new doctrine/property-accessor package.